### ISSUE TITLE
PHP 8.1 compatibility issue

### ISSUE THREAD

**Issue by novikor** [klaviyo/magento2-klaviyo#209]
```
Exception thrown with message "Deprecated Functionality: number_format(): Passing null to parameter #1 ($num) of type float is deprecated in /var/www/html/vendor/klaviyo/magento2-extension/Block/Catalog/Product/ViewedProduct.php on line 135"

Stacktrace:
#75 Exception in /var/www/html/vendor/magento/framework/App/ErrorHandler.php:62
#74 Magento\Framework\App\ErrorHandler:handler in [internal]:0
#73 number_format in /var/www/html/vendor/klaviyo/magento2-extension/Block/Catalog/Product/ViewedProduct.php:135
#72 Klaviyo\Reclaim\Block\Catalog\Product\ViewedProduct:getPrice in /var/www/html/vendor/klaviyo/magento2-extension/Block/Catalog/Product/ViewedProduct.php:185
#71 Klaviyo\Reclaim\Block\Catalog\Product\ViewedProduct:getViewedProductJson in /var/www/html/vendor/klaviyo/magento2-extension/view/frontend/templates/product/viewed.phtml:8
#70 include in /var/www/html/vendor/magento/framework/View/TemplateEngine/Php.php:71
#69 Magento\Framework\View\TemplateEngine\Php:render in /var/www/html/vendor/magento/framework/Interception/Interceptor.php:58
#68 Magento\Framework\View\TemplateEngine\Php\Interceptor:___callParent in /var/www/html/vendor/magento/framework/Interception/Interceptor.php:138
#67 Magento\Framework\View\TemplateEngine\Php\Interceptor:Magento\Framework\Interception\{closure} in /var/www/html/vendor/magento/framework/Interception/Interceptor.php:153
#66 Magento\Framework\View\TemplateEngine\Php\Interceptor:___callPlugins in /var/www/html/generated/code/Magento/Framework/View/TemplateEngine/Php/Interceptor.php:23
#65 Magento\Framework\View\TemplateEngine\Php\Interceptor:render in /var/www/html/vendor/magento/framework/View/Element/Template.php:263
#64 Magento\Framework\View\Element\Template:fetchView in /var/www/html/vendor/magento/framework/View/Element/Template.php:293
#63 Magento\Framework\View\Element\Template:_toHtml in /var/www/html/vendor/klaviyo/magento2-extension/Block/Catalog/Product/ViewedProduct.php:170
#62 Klaviyo\Reclaim\Block\Catalog\Product\ViewedProduct:_toHtml in /var/www/html/vendor/magento/framework/View/Element/AbstractBlock.php:1095
#61 Magento\Framework\View\Element\AbstractBlock:Magento\Framework\View\Element\{closure} in /var/www/html/vendor/magento/framework/View/Element/AbstractBlock.php:1099
#60 Magento\Framework\View\Element\AbstractBlock:_loadCache in /var/www/html/vendor/magento/framework/View/Element/AbstractBlock.php:660
#59 Magento\Framework\View\Element\AbstractBlock:toHtml in /var/www/html/vendor/magento/framework/View/Layout.php:578
#58 Magento\Framework\View\Layout:_renderBlock in /var/www/html/vendor/magento/framework/View/Layout.php:555
#57 Magento\Framework\View\Layout:renderNonCachedElement in /var/www/html/generated/code/Magento/Framework/View/Layout/Interceptor.php:149
#56 Magento\Framework\View\Layout\Interceptor:renderNonCachedElement in /var/www/html/vendor/magento/framework/View/Layout.php:510
#55 Magento\Framework\View\Layout:renderElement in /var/www/html/generated/code/Magento/Framework/View/Layout/Interceptor.php:140
#54 Magento\Framework\View\Layout\Interceptor:renderElement in /var/www/html/vendor/magento/framework/View/Layout.php:606
#53 Magento\Framework\View\Layout:_renderContainer in /var/www/html/vendor/magento/framework/View/Layout.php:557
#52 Magento\Framework\View\Layout:renderNonCachedElement in /var/www/html/generated/code/Magento/Framework/View/Layout/Interceptor.php:149
#51 Magento\Framework\View\Layout\Interceptor:renderNonCachedElement in /var/www/html/vendor/magento/framework/View/Layout.php:510
#50 Magento\Framework\View\Layout:renderElement in /var/www/html/generated/code/Magento/Framework/View/Layout/Interceptor.php:140
#49 Magento\Framework\View\Layout\Interceptor:renderElement in /var/www/html/vendor/magento/framework/View/Layout.php:606
#48 Magento\Framework\View\Layout:_renderContainer in /var/www/html/vendor/magento/framework/View/Layout.php:557
#47 Magento\Framework\View\Layout:renderNonCachedElement in /var/www/html/generated/code/Magento/Framework/View/Layout/Interceptor.php:149
#46 Magento\Framework\View\Layout\Interceptor:renderNonCachedElement in /var/www/html/vendor/magento/framework/View/Layout.php:510
#45 Magento\Framework\View\Layout:renderElement in /var/www/html/generated/code/Magento/Framework/View/Layout/Interceptor.php:140
#44 Magento\Framework\View\Layout\Interceptor:renderElement in /var/www/html/vendor/magento/framework/View/Layout.php:606
#43 Magento\Framework\View\Layout:_renderContainer in /var/www/html/vendor/magento/framework/View/Layout.php:557
#42 Magento\Framework\View\Layout:renderNonCachedElement in /var/www/html/generated/code/Magento/Framework/View/Layout/Interceptor.php:149
#41 Magento\Framework\View\Layout\Interceptor:renderNonCachedElement in /var/www/html/vendor/magento/framework/View/Layout.php:510
#40 Magento\Framework\View\Layout:renderElement in /var/www/html/generated/code/Magento/Framework/View/Layout/Interceptor.php:140
#39 Magento\Framework\View\Layout\Interceptor:renderElement in /var/www/html/vendor/magento/framework/View/Layout.php:606
#38 Magento\Framework\View\Layout:_renderContainer in /var/www/html/vendor/magento/framework/View/Layout.php:557
#37 Magento\Framework\View\Layout:renderNonCachedElement in /var/www/html/generated/code/Magento/Framework/View/Layout/Interceptor.php:149
#36 Magento\Framework\View\Layout\Interceptor:renderNonCachedElement in /var/www/html/vendor/magento/framework/View/Layout.php:510
#35 Magento\Framework\View\Layout:renderElement in /var/www/html/generated/code/Magento/Framework/View/Layout/Interceptor.php:140
#34 Magento\Framework\View\Layout\Interceptor:renderElement in /var/www/html/vendor/magento/framework/View/Layout.php:606
#33 Magento\Framework\View\Layout:_renderContainer in /var/www/html/vendor/magento/framework/View/Layout.php:557
#32 Magento\Framework\View\Layout:renderNonCachedElement in /var/www/html/generated/code/Magento/Framework/View/Layout/Interceptor.php:149
#31 Magento\Framework\View\Layout\Interceptor:renderNonCachedElement in /var/www/html/vendor/magento/framework/View/Layout.php:510
#30 Magento\Framework\View\Layout:renderElement in /var/www/html/generated/code/Magento/Framework/View/Layout/Interceptor.php:140
#29 Magento\Framework\View\Layout\Interceptor:renderElement in /var/www/html/vendor/magento/framework/View/Layout.php:606
#28 Magento\Framework\View\Layout:_renderContainer in /var/www/html/vendor/magento/framework/View/Layout.php:557
#27 Magento\Framework\View\Layout:renderNonCachedElement in /var/www/html/generated/code/Magento/Framework/View/Layout/Interceptor.php:149
#26 Magento\Framework\View\Layout\Interceptor:renderNonCachedElement in /var/www/html/vendor/magento/framework/View/Layout.php:510
#25 Magento\Framework\View\Layout:renderElement in /var/www/html/generated/code/Magento/Framework/View/Layout/Interceptor.php:140
#24 Magento\Framework\View\Layout\Interceptor:renderElement in /var/www/html/vendor/magento/framework/View/Layout.php:606
#23 Magento\Framework\View\Layout:_renderContainer in /var/www/html/vendor/magento/framework/View/Layout.php:557
#22 Magento\Framework\View\Layout:renderNonCachedElement in /var/www/html/generated/code/Magento/Framework/View/Layout/Interceptor.php:149
#21 Magento\Framework\View\Layout\Interceptor:renderNonCachedElement in /var/www/html/vendor/magento/framework/View/Layout.php:510
#20 Magento\Framework\View\Layout:renderElement in /var/www/html/generated/code/Magento/Framework/View/Layout/Interceptor.php:140
#19 Magento\Framework\View\Layout\Interceptor:renderElement in /var/www/html/vendor/magento/framework/View/Layout.php:975
#18 Magento\Framework\View\Layout:getOutput in /var/www/html/vendor/magento/framework/Interception/Interceptor.php:58
#17 Magento\Framework\View\Layout\Interceptor:___callParent in /var/www/html/vendor/magento/framework/Interception/Interceptor.php:138
#16 Magento\Framework\View\Layout\Interceptor:Magento\Framework\Interception\{closure} in /var/www/html/vendor/magento/framework/Interception/Interceptor.php:153
#15 Magento\Framework\View\Layout\Interceptor:___callPlugins in /var/www/html/generated/code/Magento/Framework/View/Layout/Interceptor.php:347
#14 Magento\Framework\View\Layout\Interceptor:getOutput in /var/www/html/vendor/magento/framework/View/Result/Page.php:260
#13 Magento\Framework\View\Result\Page:render in /var/www/html/vendor/magento/framework/View/Result/Layout.php:171
#12 Magento\Framework\View\Result\Layout:renderResult in /var/www/html/vendor/magento/framework/Interception/Interceptor.php:58
#11 Magento\Framework\View\Result\Page\Interceptor:___callParent in /var/www/html/vendor/magento/framework/Interception/Interceptor.php:138
#10 Magento\Framework\View\Result\Page\Interceptor:Magento\Framework\Interception\{closure} in /var/www/html/vendor/magento/framework/Interception/Interceptor.php:153
#9 Magento\Framework\View\Result\Page\Interceptor:___callPlugins in /var/www/html/generated/code/Magento/Framework/View/Result/Page/Interceptor.php:95
#8 Magento\Framework\View\Result\Page\Interceptor:renderResult in /var/www/html/vendor/magento/framework/App/Http.php:120
#7 Magento\Framework\App\Http:launch in /var/www/html/vendor/magento/framework/Interception/Interceptor.php:58
#6 Magento\Framework\App\Http\Interceptor:___callParent in /var/www/html/vendor/magento/framework/Interception/Interceptor.php:138
#5 Magento\Framework\App\Http\Interceptor:Magento\Framework\Interception\{closure} in /var/www/html/vendor/yireo/magento2-whoops/Plugin/HttpApp.php:102
#4 Yireo\Whoops\Plugin\HttpApp:aroundLaunch in /var/www/html/vendor/magento/framework/Interception/Interceptor.php:135
#3 Magento\Framework\App\Http\Interceptor:Magento\Framework\Interception\{closure} in /var/www/html/vendor/magento/framework/Interception/Interceptor.php:153
#2 Magento\Framework\App\Http\Interceptor:___callPlugins in /var/www/html/generated/code/Magento/Framework/App/Http/Interceptor.php:23
#1 Magento\Framework\App\Http\Interceptor:launch in /var/www/html/vendor/magento/framework/App/Bootstrap.php:264
#0 Magento\Framework\App\Bootstrap:run in /var/www/html/pub/index.php:30
```

**Comment by nabintiw** [comment]
We are currently investigating the issue.

**Comment by cykolln** [comment]
Hi @novikor can you provide some more details as to how to recreate this issue - I understand we need to conditionally call `number_format`, but is this happening on every product page? Or just products with a certain configuration

**Comment by novikor** [comment]
@cykolln thank you for working on the issue. Unfortunately, I don't remember to be sure - but I believe it was on every product page, simple or bundle.

I have just applied a patch with null to float conversion as a hotfix

**Comment by cykolln** [comment]
The fix for this has been merged and will be included in the next release, I will update here once the new version is out - it should be within the next week.

**Comment by cykolln** [comment]
This has been fixed as of the latest release 4.0.7
